### PR TITLE
feat: add web3 tab at instagram

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/icons/general/Web3Tab.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createIcon } from '../utils'
+
+export const Web3TabIcon = createIcon(
```
Comment (guanbinrui) [inline packages/icons/general/Web3Tab.tsx]: Rename this icon name to `CollectibleIcon`.

File: packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+function getStyleProps() {
+    const EMPTY_STYLE = {} as CSSStyleDeclaration
+
+    const eleTab = searchProfileTabSelector().evaluate() as Element
+    const style = eleTab ? window.getComputedStyle(eleTab) : EMPTY_STYLE
+
+    const activeTab = searchProfileActiveTabSelector().evaluate() as Element
+    console.log(activeTab)
+    const activeStyle = activeTab ? window.getComputedStyle(activeTab) : EMPTY_STYLE
+    console.log(activeStyle.color)
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx]:
```suggestion
    const activeTab = searchProfileActiveTabSelector().evaluate() as Element
    const activeStyle = activeTab ? window.getComputedStyle(activeTab) : EMPTY_STYLE
```

**Review by guanbinrui — DISMISSED** [review]

**Comment by guanbinrui** [comment]
@DimensionDev/qa

File: packages/mask/src/social-network-adaptor/instagram.com/collecting/identity.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        const handle = getUserId()
+        const avatar = getAvatar()
+
+        ref.value = {
+            identifier: new ProfileIdentifier(instagramBase.networkIdentifier, handle),
+            nickname,
+            avatar,
+            bio,
+        }
+        Services.Helper.resolveTCOLink(homepage).then((link) => {
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/collecting/identity.ts]: wait, Instagram is also using `t.co` short link? I thought it was only for Twitter

File: packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    const contentWatcher = new MutationObserverWatcher(searchProfileTabPageSelector()).useForeach(() => {
+        const elePage = searchProfileTabPageSelector().evaluate()
+        if (elePage && !tabInjected) {
+            const watcher = new MutationObserverWatcher(searchProfileTabListLastChildSelector())
+            startWatch(watcher, signal)
+            createReactRootShadowed(watcher.firstDOMProxy.afterShadow, { signal }).render(<ProfileTabAtInstagram />)
+            tabInjected = true
+        }
+    })
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx]:
```suggestion
const contentWatcher = new MutationObserverWatcher(searchProfileTabPageSelector()).useForeach((node, key, proxy) => {
    const subController = new AbortController()
    const subSignal = combineAbortSignal(signal, subController.signal)
    const elePage = proxy.realCurrent
    if (elePage && !tabInjected) {
        const watcher = new MutationObserverWatcher(searchProfileTabListLastChildSelector())
        startWatch(watcher, subSignal)
        createReactRootShadowed(watcher.firstDOMProxy.afterShadow, { signal: subSignal }).render(<ProfileTabAtInstagram />)
        tabInjected = true
    }
    return () => subController.abort()
})
```

File: packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+    startWatch(contentWatcher, signal)
+}
+
+function getStyleProps() {
+    const EMPTY_STYLE = {} as CSSStyleDeclaration
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx]:
```suggestion
    const EMPTY_STYLE: CSSStyleDeclaration = {}
```

File: packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    const eleTab = searchProfileTabSelector().evaluate() as Element
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx]:
```suggestion
    const eleTab: Element = searchProfileTabSelector().evaluate()
```

File: packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    const style = eleTab ? window.getComputedStyle(eleTab) : EMPTY_STYLE
+
+    const activeTab = searchProfileActiveTabSelector().evaluate() as Element
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx]:
```suggestion
    const activeTab: Element = searchProfileActiveTabSelector().evaluate()
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+}
+const querySelectorAll = <T extends E>(selector: string) => {
+    return new LiveSelector().querySelectorAll<T>(selector)
+}
+
+export const searchProfileTabListLastChildSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[id="react-root"] section main  div[role="tablist"] > :last-child')
+
+export const searchProfileTabPageSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[id="react-root"] section main > div > div:last-child')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const searchProfileTabPageSelector: () => LiveSelector<HTMLDivElement, true> = () =>
    querySelector<HTMLDivElement>('[id="react-root"] section main > div > div:last-child')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+export const searchProfileTabPageSelector1: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[id="react-root"] section main > div > div:last-child > :last-child')
+
+export const searchProfileTabSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[id="react-root"] section main div[role="tablist"]  a[aria-selected="false"]')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const searchProfileTabSelector: () => LiveSelector<HTMLAnchorElement, true> = () =>
    querySelector<HTMLAnchorElement>('[id="react-root"] section main div[role="tablist"]  a[aria-selected="false"]')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+export const searchProfileActiveTabSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[id="react-root"] section main  div[role="tablist"] a[aria-selected="true"]')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const searchProfileActiveTabSelector: () => LiveSelector<HTMLAnchorElement, true> = () =>
    querySelector<HTMLAnchorElement>('[id="react-root"] section main  div[role="tablist"] a[aria-selected="true"]')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+export const searchAvatarSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[id="react-root"] section main header button  img')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const searchAvatarSelector: () => LiveSelector<HTMLImageElement, true> = () =>
    querySelector<HTMLImageElement>('[id="react-root"] section main header button  img')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+export const bioDescriptionSelector = () =>
+    querySelector<HTMLDivElement>('[id="react-root"] section main header section > div:last-child > :nth-child(3)')
+
+export const personalHomepageSelector = () =>
+    querySelector<HTMLDivElement>('[id="react-root"] section main header section > div:last-child  a')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const personalHomepageSelector = () =>
    querySelector<HTMLAnchorElement>('[id="react-root"] section main header section > div:last-child  a')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+export const searchNickNameSelector = () =>
+    querySelector<HTMLDivElement>('[id="react-root"] section main header section > div:last-child > span')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const searchNickNameSelector = () =>
    querySelector<HTMLSpanElement>('[id="react-root"] section main header section > div:last-child > span')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+
+export const searchUserIdSelector = () =>
+    querySelector<HTMLDivElement>('[id="react-root"] section main header section h2')
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
export const searchUserIdSelector = () =>
    querySelector<HTMLHeadingElement>('[id="react-root"] section main header section h2')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/user.ts [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    return node ? collectNodeText(node) : ''
+}
+
+export const getUserId = () => {
+    const node = searchUserIdSelector().evaluate()
+    return node ? collectNodeText(node) : ''
+}
+
+export const getAvatar = () => {
+    const node = searchAvatarSelector().evaluate() as HTMLImageElement
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/user.ts]:
```suggestion
    const node: HTMLImageElement = searchAvatarSelector().evaluate()
```

File: packages/mask/src/plugins/NextID/base.ts [hunk 16]
```diff
@@ -9,7 +9,7 @@ export const base: Plugin.Shared.Definition = {
     publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
     enableRequirement: {
         architecture: { app: false, web: true },
-        networks: { type: 'opt-out', networks: {} },
+        networks: { type: 'opt-out', networks: { [CurrentSNSNetwork.Instagram]: true } },
```
Comment (Jack-Works) [inline packages/mask/src/plugins/NextID/base.ts]: does next id work on Facebook and minds?

Comment (guanbinrui) [reply packages/mask/src/plugins/NextID/base.ts]: Yes, it should work on fb and minds.

File: packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+        })
+    }
+    const clear = () => {
+        const style = getStyleProps()
+        const activeTab = searchProfileActiveTabSelector().evaluate()
+        if (activeTab?.style) {
+            activeTab.style.borderTop = 'none'
+            activeTab.style.color = style.color
+        }
+        Array.from(searchProfileTabPageSelector().evaluate()?.childNodes ?? []).map((v) => {
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/web3tab/ProfileTab.tsx]:
```suggestion
        Array.from(searchProfileTabPageSelector().evaluate()?.childNodes ?? []).forEach((v) => {
```

File: packages/mask/src/plugins/NextID/base.ts [hunk 18]
```diff
@@ -9,7 +9,14 @@ export const base: Plugin.Shared.Definition = {
     publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
     enableRequirement: {
         architecture: { app: false, web: true },
-        networks: { type: 'opt-out', networks: {} },
+        networks: {
+            type: 'opt-out',
```
Comment (Jack-Works) [inline packages/mask/src/plugins/NextID/base.ts]: suggest use `opt-in` and opt-in for twitter and facebook.

**Comment by wenluomask** [comment]
Refer to Jira: https://mask.atlassian.net/browse/MASK-749

File: packages/mask/src/components/InjectedComponents/PageTab.tsx [hunk 19]
```diff
@@ -20,6 +21,13 @@ export function PageTab(props: PageTabProps) {
     const { tabs, selectedTab, onChange } = props
     const { classes } = useStyles()
 
+    if (!tabs.length) {
+        return (
+            <Typography variant="body2" color="textPrimary">
+                No address found
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/PageTab.tsx]: `useI18N()`

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 20]
```diff
@@ -0,0 +1,44 @@
+import { LiveSelector } from '@dimensiondev/holoflows-kit'
+
+type E = HTMLDivElement
```
Comment (nuanyang233) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
type E = HTMLELement
```

**Comment by wenluomask** [comment]
Checked.

**Review by guanbinrui — DISMISSED** [review]

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+    querySelector<E>('[id="react-root"] section main > div > div:last-child')
+
+export const searchProfileTabSelector = () =>
+    querySelector<E>('[id="react-root"] section main div[role="tablist"]  a[aria-selected="false"]')
+
+export const searchProfileActiveTabSelector = () =>
+    querySelector<E>('[id="react-root"] section main  div[role="tablist"] a[aria-selected="true"]')
+
+export const searchAvatarSelector = () =>
+    querySelector<HTMLImageElement>('[id="react-root"] section main header button  img')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
    querySelector<HTMLImageElement>('[id="react-root"] section main header button img')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+
+export const bioDescriptionSelector = () =>
+    querySelector<E>('[id="react-root"] section main header section > div:last-child div')
+
+export const personalHomepageSelector = () =>
+    querySelector<E>('[id="react-root"] section main header section > div:last-child  a')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
    querySelector<E>('[id="react-root"] section main header section > div:last-child a')
```

File: packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+export const searchProfileTabListLastChildSelector = () =>
+    querySelector<E>('[id="react-root"] section main  div[role="tablist"] > :last-child')
+
+export const searchProfileTabPageSelector = () =>
+    querySelector<E>('[id="react-root"] section main > div > div:last-child')
+
+export const searchProfileTabSelector = () =>
+    querySelector<E>('[id="react-root"] section main div[role="tablist"]  a[aria-selected="false"]')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/instagram.com/utils/selector.ts]:
```suggestion
    querySelector<E>('[id="react-root"] section main div[role="tablist"] a[aria-selected="false"]')
```

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — APPROVED** [review]

File: packages/mask/src/social-network-adaptor/instagram.com/injection/ProfileTab.tsx [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+        if (activeTab?.style) {
+            activeTab.style.borderTop = 'none'
+            activeTab.style.color = style.color
+        }
+        // hide the content page
+        await untilElementAvailable(searchProfileTabPageSelector())
+
+        Array.from(searchProfileTabPageSelector().evaluate()?.childNodes ?? []).forEach((v) => {
+            const ele = v as HTMLDivElement
+            if (ele.tagName !== 'SPAN') ele.style.display = 'none'
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/instagram.com/injection/ProfileTab.tsx:107]: ????

**Pull request merged**